jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/SchemaRule.java

```diff
@@ -63,7 +63,7 @@ protected SchemaRule(RuleFactory ruleFactory) {
     @Override
     public JType apply(String nodeName, JsonNode schemaNode, JsonNode parent, JClassContainer generatableType, Schema schema) {
 
-        if (schemaNode.has("$ref")) {
+        if (schemaNode.has("$ref") && !schemaNode.has("existingJavaType")) {
```

Comment:
Can it be that schema node would have both `$ref` and `existingJavaType` ?
Perhaps a test case for such scenario could be added

Reply:
I will add a test.
I am not sure however of the desired behavior when it comes to having the root object in the schema with a $ref (and could not find a test for that).
Today, there is no class generated, other than the class corresponding to the referenced schema. Is that the desired behavior @unkish ?

Reply:
> I am not sure however of the desired behavior when it comes to having the root object in the schema with a $ref (and could not find a test for that).

You're right - at present there is no test for that.

> Today, there is no class generated, other than the class corresponding to the referenced schema. Is that the desired behavior ?

It seems that it is the desired behavior as it is how the code currently works (perhaps @joelittlejohn could prove that assumption to be wrong)

Reply:
In the past, the JSON Schema specification dictated that when $ref is used, all other properties *must* be ignored. The referenced schema replaces the schema node in its entirety. This was a rule in the spec up to (and including) draft-07.

For schema release '2019-09' (a.k.a. draft-08) this rule was dropped from the JSON Schema specification.

I think it's useful to allow `existingJavaType` to be used in this way. Now that it is not explicitly disallowed by the spec, I'm happy to make this change. It's a minor breaking change, it's true, but unlikely to affect anyone tbh. I don't like doing it, but it's pretty useful.

Reply:
There is a strong case to allow `javaType` too IMO.